src/core/lv_obj_tree.c

```diff
@@ -202,22 +202,28 @@ void lv_obj_move_to_index(lv_obj_t * obj, int32_t index)
 {
     LV_ASSERT_OBJ(obj, MY_CLASS);
 
+    /* Check parent validity */
     lv_obj_t * parent = lv_obj_get_parent(obj);
-
     if(!parent) {
         LV_LOG_WARN("parent is NULL");
         return;
     }
 
+    const uint32_t parent_child_count = lv_obj_get_child_count(parent);
+    const int32_t old_index = lv_obj_get_index(obj);
+
     if(index < 0) {
-        index = lv_obj_get_child_count(parent) + index;
+        index += parent_child_count;
     }
 
-    const int32_t old_index = lv_obj_get_index(obj);
+    if((index < 0)
```

Comment:
When this condition can be TRUE? If the `index < 0` is TRUE above, this is not possible, isn't?

Reply:
Can be `-1` on error 
- [here](https://github.com/lvgl/lvgl/blob/master/src/core/lv_obj_tree.c#L412) (should be `return -1`)
- or [here](https://github.com/lvgl/lvgl/blob/master/src/core/lv_obj_tree.c#L419)

Reply:
So, should I replace the return value (0xFFFFFFFF) with -1?